### PR TITLE
common date parsing

### PR DESCRIPTION
**Pull request by rigelrozanski**
parsing dates and date ranges

### CODE REVIEW

File: common/date.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// ParseDate parses a date string of the format YYYY-MM-DD
+func ParseDate(date string) (t time.Time, err error) {
```
Comment (melekes) [inline common/date.go]: aren't there any existing functions? https://play.golang.org/p/0Q1ChGzGwJ

Comment (rigelrozanski) [inline common/date.go]: I had a bunch of trouble getting this to work... really confusing with the month day being represented by the 02 and 01 respectively. BUT THANKS. this makes sense... Do we still want to include this library for date range parsing? Also nice to just have the layout predefined here I think... I'll update the code

Comment (rigelrozanski) [inline common/date.go]: like why on earth is the layout have to be January 2, 2006?? haha

Comment (rigelrozanski) [inline common/date.go]: okay update here

File: common/date.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import (
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// ParseDate parses a date string of the format YYYY-MM-DD
+func ParseDate(date string) (t time.Time, err error) {
+	layout := "2006-01-02" //this represents YYYY-MM-DD
```
Comment (melekes) [inline common/date.go]: why not a constant? then we could remove this func altogether

Comment (rigelrozanski) [inline common/date.go]: This function is used outside of the ParseDateRange function too, so it should probably exist. We can make layout a constant but I don't think it makes a difference, your call given that this function needs to exist either way

Comment (melekes) [inline common/date.go]: > This function is used outside of the ParseDateRange function too

where exactly?

Comment (rigelrozanski) [inline common/date.go]: I'm using in trackomatron

Comment (ebuchman) [inline common/date.go]: If we make the layout an exposed constant, others can just call `time.Parse(cmn.TimeLayour, date)`, and then we don't need this function. I don't mind it too much though

File: common/date_test.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	}{
+		{"2015-12-31", date, true},
+		{"2015-31-12", date, false},
+		{"12-31-2015", date, false},
+		{"31-12-2015", date, false},
+	}
+
+	for _, test := range testDates {
+		parsed, err := ParseDate(test.dateStr)
+		switch test.errNil {
```
Comment (ebuchman) [inline common/date_test.go]: No need for a switch/case if there's only two cases. If statement is better

File: common/date.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	return time.Parse(layout, date)
+}
+
+// ParseDateRange parses a date range string of the format start:end
+//   where the start and end date are of the format YYYY-MM-DD.
+//   The parsed dates are *time.Time and will return nil pointers for
```
Comment (ebuchman) [inline common/date.go]: Why do we return pointers? Is it so we can use nil? Is the zero time and IsZero not sufficient?

**Comment by ebuchman** [comment]
Changelog please

File: common/date.go (hunk 2 above)
Comment (rigelrozanski) [inline common/date.go]: Okay, don't really mind, implemented ebuchman's suggestion

File: common/date_test.go (hunk 3 above)
Comment (rigelrozanski) [inline common/date_test.go]: Thanks, changed

File: common/date.go (hunk 4 above)
Comment (rigelrozanski) [inline common/date.go]: Oh great, yes, I agree that IsZero makes more sense

**Pull request merged**